model/fn-execution/src/main/proto/beam_provision_api.proto

```diff
@@ -67,6 +67,10 @@ message ProvisionInfo {
     // (optional) Resource limits that the SDK harness worker should respect.
     // Runners may -- but are not required to -- enforce any limits provided.
     Resources resource_limits = 4;
+
+    // (required) The artifact retrieval token produced by
```

Comment:
Nit: this token should not have to be identical to the commit side.

Reply:
(addressed by previous comment: this token is, by definition, a way to pass information from the Commit call to the retrieval service - so it has to be identical to the commit side)

Reply:
It's not a requirement that the same underlying storage and necessary metadata is used for staging and retrieval (which might use a caching intermediate, say), so this seems like an unnecessary restriction. The user/sdk-side code may run in completely different environments.

Reply:
I think a caching intermediate would not need its own token format - it could just pass the given token to the "real" retrieval service?..